index.js

```diff
@@ -280,6 +293,179 @@ app.get('/playlists', (request, response) => {
+
+            pool.query(queryText, newUser, (err, result) => {
+                if (err) {
+                    console.error('query error:', err.stack);
+                    response.send('query error');
+                }
+                // send login cookies
+                console.log("Registered" + result.rows[0]);
+                response.cookie('user_name', result.rows[0].name);
+                response.cookie('loggedIn', result.rows[0].password);
+                // redirect to homepage
+                response.redirect('/artists');
+            });
+        }
+    });
+});
+
+app.get('/login', (request, response) => {
+  // respond with HTML page with form to register
+  let output = {};
+  output.message = "Login Account";
+  response.render('account', output);
+});
+
+app.post('/login', (request, response) => {
+    // check user login
+    let user_name = request.body.name;
+    let hashedPw = sha256(request.body.password + SALT);
+
+    let queryText = `SELECT * FROM users WHERE name='${user_name}'`;
+
+    pool.query(queryText, (err, result) => {
+        if (err) {
+            console.error('query error:', err.stack);
+            response.send('query error');
+        }
+        // if there is result
+        if ( result.rows.length > 0 ) {
+            // check if password correct
+            if ( hashedPw === result.rows[0].password) {
+                // send login cookies
+                response.cookie('user_id', result.rows[0].id);
+                response.cookie('user_name', result.rows[0].name);
+                response.cookie('loggedIn', result.rows[0].password);
+                // redirect to homepage
+                response.redirect('/artists');
+            } else {
+                // else return incorrect password
+                let output = {};
+                output.message = "Incorrect password, please try again";
+                response.render('account', output);
+            }
+        } else {
+            // if there is no result
+            let output = {};
+            output.message = "Incorrect user name, please try again";
+            response.render('account', output);
+        }
+    });
+});
+
+app.get('/favorites/new', (request, response) => {
+    // respond with HTML page with form to add new song to favorites
+    // get user_name from cookie
+    let user_name = request.cookies['user_name'];
+    // check if password correct?
+    console.log("Adding to favorites");
+    // input songs list to be rendered as option
+    let queryText = 'SELECT * FROM songs';
+
+    pool.query(queryText, (err, result) => {
+        if (err) {
+            console.error('query error:', err.stack);
+            response.send('query error');
+        }
+        let list = {};
+        list.message = "Add Song to " + user_name + " Favorites";
+        list.songs = result.rows;
+        response.render('newlistsong', list);
+    });
+});
+
+app.post('/favorites/new', (request, response) => {
```

Comment:
try to follow the restful routes, usually app.post for new element eg favorites will be "/favourites" without the "new"